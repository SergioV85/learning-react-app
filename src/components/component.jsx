import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}