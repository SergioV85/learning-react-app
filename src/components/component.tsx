import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}