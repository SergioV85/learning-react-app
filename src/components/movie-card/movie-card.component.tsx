import * as React from 'react';
import { DateTime } from 'luxon';

export interface MovieObject {
  budget: number;
  genres: string[];
  id: number;
  overview: string;
  poster_path: string;
  release_date: string;
  revenue: number;
  runtime: number;
  tagline: string;
  title: string;
  vote_average: number;
  vote_count: number;
}

interface MovieCardProps {
  movie: MovieObject;
  onMovieClick(id: number): void;
}

interface MovieCardState {}

export class MovieCard extends React.Component<MovieCardProps, MovieCardState> {
  constructor(props: MovieCardProps) {
    super(props);
  }

  public render() {
    return <div className="c-movie-card" onClick={() => this.props.onMovieClick(this.props.movie.id)}>
      <img className="c-movie-card__poster" src={this.props.movie.poster_path} />
      <div className="c-movie-card__title-and-year">
        <span className="c-movie-card__title">{this.props.movie.title}</span>
        <span className="c-movie-card__year">
          { DateTime.fromISO(this.props.movie.release_date).year }
        </span>
      </div>
      <div className="c-movie-card__genres">
        { this.props.movie.genres.join(', ') }
      </div>
    </div>;
  }
}