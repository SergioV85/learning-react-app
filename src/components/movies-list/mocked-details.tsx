export const mockedMovieDetails = {
  "budget": 0,
  "id": 419831,
  "overview": "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
  "poster_path": "https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg",
  "release_date": "2018-01-21",
  "revenue": 0,
  "runtime": 106,
  "tagline": "",
  "title": "I Kill Giants",
  "vote_average": 6,
  "vote_count": 63,
  "genres": [
    "Thriller",
    "Drama",
    "Fantasy"
  ],
}