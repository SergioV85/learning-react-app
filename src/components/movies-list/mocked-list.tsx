export const mockedMoviesList = [{
    "id": 419831,
    "title": "I Kill Giants",
    "tagline": "",
    "vote_average": 6,
    "vote_count": 63,
    "release_date": "2018-01-21",
    "poster_path": "https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg",
    "overview": "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
    "budget": 0,
    "revenue": 0,
    "genres": [
      "Thriller",
      "Drama",
      "Fantasy"
    ],
    "runtime": 106
  },
  {
    "id": 399057,
    "title": "The Killing of a Sacred Deer",
    "tagline": "",
    "vote_average": 6.6,
    "vote_count": 509,
    "release_date": "2017-10-20",
    "poster_path": "https://image.tmdb.org/t/p/w500/e4DGlsc9g0h5AyoyvvAuIRnofN7.jpg",
    "overview": "A teenager's attempts to bring a brilliant surgeon into his dysfunctional family take an unexpected turn.",
    "budget": 0,
    "revenue": 2205074,
    "genres": [
      "Drama",
      "Horror",
      "Thriller"
    ],
    "runtime": 121
  },
  {
    "id": 423988,
    "title": "Kill Switch",
    "tagline": "Worlds will collide",
    "vote_average": 5,
    "vote_count": 81,
    "release_date": "2017-06-01",
    "poster_path": "https://image.tmdb.org/t/p/w500/8K6pfMcT3hiIlCFw2RrHoXLshxl.jpg",
    "overview": "A pilot battles to save his family and the planet after an experiment for unlimited energy goes wrong.",
    "budget": 1000000,
    "revenue": 0,
    "genres": [
      "Science Fiction"
    ],
    "runtime": 91
  },
  {
    "id": 382322,
    "title": "Batman: The Killing Joke",
    "tagline": "The madness begins.",
    "vote_average": 6.2,
    "vote_count": 618,
    "release_date": "2016-07-21",
    "poster_path": "https://image.tmdb.org/t/p/w500/zm0ODjtfJfJW0W269LqsQl5OhJ8.jpg",
    "overview": "As Batman hunts for the escaped Joker, the Clown Prince of Crime attacks the Gordon family to prove a diabolical point mirroring his own fall into madness. Based on the graphic novel by Alan Moore and Brian Bolland.",
    "budget": 3500000,
    "revenue": 3775000,
    "genres": [
      "Action",
      "Animation",
      "Crime",
      "Drama"
    ],
    "runtime": 72
  },
  {
    "id": 237756,
    "title": "Kill Me Three Times",
    "tagline": "Once is never enough.",
    "vote_average": 5.5,
    "vote_count": 203,
    "release_date": "2015-04-10",
    "poster_path": "https://image.tmdb.org/t/p/w500/zqTlUUBlVZTzVMYUnJQve0kJc3o.jpg",
    "overview": "While on a seemingly routine job, a jaded hit man discovers that he's not the only one with his target in the crosshairs.",
    "budget": 0,
    "revenue": 0,
    "genres": [
      "Comedy",
      "Thriller"
    ],
    "runtime": 90
  }
]