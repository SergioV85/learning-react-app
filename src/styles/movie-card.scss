.c-movie-card {
  height: 350px;
  margin: 10px 20px;
  width: 200px;
  
  &__poster {
    height: auto;
    max-height: 300px;
    max-width: 100%;
  }
  &__title-and-year {
    align-items: flex-start;
    display: flex;
    justify-content: space-between;
    margin-top: 10px;
  }
  &__title {
    color: $primary-dark;
    font-family: $main-font;
    font-size: 0.75rem;
    font-weight: 700;
    text-transform: uppercase;
  }
  &__year {
    border: 1px solid $dark-grey;
    border-radius: 2px;
    color: $primary-dark;
    font-family: $main-font;
    font-size: 0.75rem;
    padding: 0 2px;
  }
  &__genres {
    color: $primary-dark;
    font-family: $main-font;
    font-size: 0.65rem;
  }
}