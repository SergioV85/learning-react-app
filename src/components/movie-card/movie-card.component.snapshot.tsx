import * as React from 'react';
import { shallow } from 'enzyme';
import { MovieCard } from './movie-card.component';
import { mockedMovieDetails } from './../movies-list/mocked-details';

describe('Footer Component Snapshot', () => {
  it('should renders correctly', () => {
    const movieCardSnapshot = shallow(
      <MovieCard movie={mockedMovieDetails} onMovieClick={() => {}} />
    );
    expect(movieCardSnapshot)
      .toMatchSnapshot();
  });
});