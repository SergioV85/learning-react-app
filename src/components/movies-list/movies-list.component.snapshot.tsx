import * as React from 'react';
import { shallow } from 'enzyme';
import { MoviesList } from './movies-list.component';
import { mockedMoviesList } from './mocked-list';

describe('Movies List Component Snapshot', () => {
  it('should renders correctly', () => {
    const moviesListSnapshot = shallow(
      <MoviesList movies={mockedMoviesList} onMovieClick={() => {}} />
    );
    expect(moviesListSnapshot)
      .toMatchSnapshot();
  });
});