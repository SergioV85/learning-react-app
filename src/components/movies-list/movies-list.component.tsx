import * as React from 'react';
import { ErrorBoundary } from './../error-boundary/error-boundary.component';
import { MovieCard, MovieObject } from './../movie-card/movie-card.component';
interface MoviesListProps {
  movies: MovieObject[];
  onMovieClick(): void;
}

interface MoviesListState {}

export class MoviesList extends React.Component<MoviesListProps, MoviesListState> {
  constructor(props: MoviesListProps) {
    super(props);
  }

  public render() {
    return <div className={`c-movies-list ${this.props.movies.length === 0 ? 'c-movies-list--empty' : ''}`}>
      {
        this.props.movies.length === 0
        ? <h1 className="c-movies-list__no-found-title">No films found</h1>
        : <ul className="c-movies-list__collection container">
            {this.props.movies.map(
              (movie) => 
                <ErrorBoundary key={`movie-card-wrapper-${movie.id}`} errorMessage={`Unfortunally, we can't show ${movie.title} preview`}>
                  <MovieCard key={movie.id} movie={movie} onMovieClick={this.props.onMovieClick} />
                </ErrorBoundary>
            )}
          </ul>
      }
    </div>;
  }
}