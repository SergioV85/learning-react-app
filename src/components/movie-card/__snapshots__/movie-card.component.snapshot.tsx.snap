// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Snapshot should renders correctly 1`] = `
<div
  className="c-movie-card"
  onClick={[Function]}
>
  <img
    className="c-movie-card__poster"
    src="https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg"
  />
  <div
    className="c-movie-card__title-and-year"
  >
    <span
      className="c-movie-card__title"
    >
      I Kill Giants
    </span>
    <span
      className="c-movie-card__year"
    >
      2018
    </span>
  </div>
  <div
    className="c-movie-card__genres"
  >
    Thriller, Drama, Fantasy
  </div>
</div>
`;
